src/mysql.c

```diff
@@ -238,6 +240,13 @@ static int mysql_config (oconfig_item_t *ci) /* {{{ */
 
 static MYSQL *getconnection (mysql_database_t *db)
 {
+	// This is necessary because we might be running on a different
+	// thread than before.
+	static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+	pthread_mutex_lock(&mutex);
+	mysql_library_init(0, NULL, NULL);
```

Comment:
That's the first thing we tried. The docs are not correct. I will add a note shortly explaining in more detail.